### PR TITLE
RFC: Getter CP readOnly by default

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by mmun** [comment]
What's the status here? Should we rebase and merge?

**Comment by stefanpenner** [comment]
pending @wycats pings

**Comment by cibernox** [comment]
Now that the new CP syntax is enabled in canary this should not take much time. I can take care of this since I already know that part of the codebase.

**Comment by rwjblue** [comment]
@cibernox - I believe that there may be some open questions (just from reading the comments above), but it should be ready for initial implementation (and that could help answer some of those questions...).

File: 0000-read-only-cp-syntax.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+firstName: Ember.computed({
+  get: function() { },
+  set: function(key, value) { this._super(key, value); }
+}); // throw new Error("ReadOnly Error blah blah blah");
+```
+
+# migration:
+
+* 1.x update internals to correctly use `writable` if needed (and runs tests against this)
```
Comment (krisselden) [inline 0000-read-only-cp-syntax.md]: @wycats this ended with "I could write it up if you want?" if you don't have time can we move forward in some way?

Comment (cibernox) [inline 0000-read-only-cp-syntax.md]: I already started to implement this. The implementation is very simple in fact, but there is loads of failing tests. It requires some thought to decide which internal properties should be writable.

**Comment by tomdale** [comment]
After discussing at the core team, we like this change but missed the boat for 2.0. Let's revisit for Ember 3.0.

**Comment by cibernox** [comment]
@tomdale I do have a local branch where I implemented this and IIRC it required very little changes. The concerns of the core team are related with this being deprecated with little notice?

**Comment by stefanpenner** [comment]
example of the future:

```js

export class extends Ember.Object {

  @computed
  otherName() {

  }

  @computed
  get fullName() {

  }
};
```

* otherName overrides.
* fullName doesn't, as isn't how es5 descriptors with missing setters work.

A future RFC will explore and propose the formal adoption of decorators in ember.

**Comment by cibernox** [comment]
I see it and I like it.

**Comment by stefanpenner** [comment]
> I see it and I like it.

:+1:

**Comment by stefanpenner** [comment]
@cibernox someone,  (maybe you or me) should write up an RFC for the future integration of decorators.

https://github.com/rwjblue/ember-computed-decorators is a fun example, but since babel fixed some things some serious issues (chaining is broken) now work. We should likely fix that soon, so we can continue to test in userland and the ES2016 spec matures.

**Pull request closed without merging**